File: Selenium/Activities/Activity10.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions

with webdriver.Firefox() as driver:
    
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")

    print(driver.title)

    wait = WebDriverWait(driver, 10)

    checkbox = driver.find_element(By.CLASS_NAME, "willDisappear")

    print("Checkbox is displayed: ", checkbox.is_displayed())

    driver.find_element(By.ID,"toggleCheckbox").click()

    wait.until(expected_conditions.invisibility_of_element_located(checkbox))

    print("Checkbox is displayed: ", checkbox.is_displayed())

    driver.close()

File: Selenium/Activities/Activity10.java
package activities;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity10 {

	public static void main(String[] args) {

		WebDriver driver =  new FirefoxDriver();
		
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		
		driver.get("https://v1.training-support.net/selenium/dynamic-controls");
		
		System.out.println(driver.getTitle());
		
		WebElement checkbox = driver.findElement(By.className("willDisappear"));
		
		System.out.println("check box is displayed: "+checkbox.isDisplayed());
		
		driver.findElement(By.id("toggleCheckbox")).click();
		
		wait.until(ExpectedConditions.invisibilityOfElementLocated(By.className("willDisappear")));
		
		System.out.println("check box is displayed: "+checkbox.isDisplayed());
		
		driver.close();

	}

}
